src/genegraph/database/query.clj

```diff
@@ -151,3 +152,19 @@ use io/slurp"
      [node1 rdf-first text]
      [node1 rdf-rest
       (NodeFactory/createURI "http://www.w3.org/1999/02/22-rdf-syntax-ns#nil")]]))
+
+(defn- node-iter->resource-set [node-iterator]
+  (->> node-iterator
+       iterator-seq
+       (filter #(and (.isResource %) (not (.isAnon %))))
+       (map resource)
+       set))
+
+(defn referenced-resources
+  "Returns a set of the resources referenced by the given model
+  (excluding predicates). Useful for expiring cache objects when
+  a model containing new objects is recieved."
+  [model]
+  (let [model-objects (node-iter->resource-set (.listObjects model))
+        model-subjects (node-iter->resource-set (.listSubjects model))]
+    (set/union model-objects model-subjects)))
```

Comment:
Maybe just this?  (I don't know how to test it though.)
```clojure
(defn referenced-resources
  "Return the set of objects and subjects referenced by MODEL (excluding
  predicates). Useful for expiring cache objects when a model
  containing new objects is received."
  [model]
  (with-open [objects  (.listObjects  model)
              subjects (.listSubjects model)]
    (->> (concat (iterator-seq objects) (iterator-seq subjects))
         (filter #(and (.isResource %) (not (.isAnon %))))
         (map resource)
         set)))
```
It looks like that grabs all objects and subjects
that are non-anonymous(?) resources --
not that I understand what that means.